CHANGELOG.md

```diff
@@ -1,3 +1,338 @@
+
+#### Security enhancements
+
+- `calico-host-protection`: Add custom locked down PSP configuration ([#1274](https://github.com/kinvolk/lokomotive/pull/1274)).
+
+#### Documentation
+
+- Add `openebs-operator` update guide ([#1163](https://github.com/kinvolk/lokomotive/pull/1163)).
+- Add `rook-ceph` update guide ([#1165](https://github.com/kinvolk/lokomotive/pull/1165)).
+
+#### Miscellaneous
+
+- Pull control plane images from Quay to avoid hitting Docker Hub pulling limits ([#1226](https://github.com/kinvolk/lokomotive/pull/1226)).
+- Bootkube now waits for all control plane charts to converge before exiting, which should make the bootstrapping process more stable ([#1085](https://github.com/kinvolk/lokomotive/pull/1085)).
+- Remove deprecated CoreOS mentions from AWS ([#1245](https://github.com/kinvolk/lokomotive/pull/1245)) and bare metal ([#1246](https://github.com/kinvolk/lokomotive/pull/1246)).
+- Improve hardware reservations validation rules on Equinix Metal ([#1186](https://github.com/kinvolk/lokomotive/pull/1186)).
+
+### Updating from v0.5.0
+
+#### Configuration syntax changes
+
+##### AWS
+
+Removed the undocumented `cluster.os_name` parameter, since Lokomotive supports Flatcar Container Linux only.
+
+##### Bare-metal
+
+The `cluster.os_channel` parameter got simplified by removing the `flatcar-` prefix.
+
+###### Old
+
+```hcl
+os_channel = "flatcar-stable"
+```
+
+###### New
+
+```hcl
+os_channel = "stable"
+```
+
+##### Velero
+
+Velero requires an explicit `provider` field to select the provider.
+Example:
+
+```hcl
+component `velero` {
+  provider = "openebs"
+
+  openebs {
+    ...
+  }
+}
+```
+
+#### Updating Prometheus Operator
+
+Upgrading this component incurs down time and some steps to be executed manually before upgrading:
+
+1. Patch the `PersistentVolume` created/used by the `prometheus-operator` component to `Retain` claim policy.
+
+  ```bash
+  kubectl patch pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-prometheus-prometheus-operator-prometheus-0")].metadata.name}')
+
+  kubectl patch pv -p '{"spec":{"persistentVolumeReclaimPolicy":"Retain"}}' $(kubectl get pv -o jsonpath='{.items[?(@.spec.claimRef.name=="data-alertmanager-prometheus-operator-alertmanager-0")].metadata.name}')
+  ```
+
+  > **Note:** To execute the above command, the user must have a cluster wide permission.
+
+2. Uninstall the `prometheus-operator` release and delete the existing `PersistentVolumeClaim`, and verify `PersistentVolume` become `Released`.
+
+  ```bash
+  lokoctl component delete prometheus-operator
+  ```
+
+  ```bash
+  kubectl delete pvc data-prometheus-prometheus-operator-prometheus-0 -n monitoring
+  kubectl delete pvc data-alertmanager-prometheus-operator-alertmanager-0 -n monitoring
+  ```
+
+  You can choose to remove all your existing CRDs (ServiceMonitors, PodMonitors, etc.) if you want to.
```

Comment:
I think we should remove this line, as we discussed.

Reply:
Yes. Also, this is pretty vague: when and why would you want to remove CRDs?

Reply:
> Yes. Also, this is pretty vague: when and why would you want to remove CRDs?

The user should not remove the CRs but rather update them. Prometheus operator has this convention of labels  it checks like if there is a certain label on configmap then that is queried for grafana dashboard, or if there is a certain label on the ServiceMonitor only then it is scraped etc. Now these labels have changed with the move to new prometheus operator chart so user is advised to udpate the components. Since these components ship the configs with new labels.